### PR TITLE
test(clickhouse): add TPC-H and TPC-DS queries

### PR DESCRIPTION
**Pull request by cpcloud**
Adds TPC-H and TPC-DS testing to clickhouse.

Depends on #9525.

### CODE REVIEW

File: ibis/backends/clickhouse/compiler.py [hunk 1]
```diff
@@ -705,3 +708,31 @@ def visit_TableUnnest(
             )
 
         return select
+
+    def _cleanup_names(
+        self, exprs: Mapping[str, sge.Expression]
+    ) -> Iterator[sge.Expression]:
+        """Compose `_gen_valid_name` and `_dedup_name` to clean up names in projections.
+
+        ClickHouse has a bug where this fails to find the final `"o"."a"` column:
```
Comment (jcrist) [inline ibis/backends/clickhouse/compiler.py:717]: Is there a link for this bug? If so, might want to add it to the docstring here as a reference.

**Review by jcrist — APPROVED** [review]
Overall LGTM!

File: ibis/backends/clickhouse/compiler.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/clickhouse/compiler.py:717]: I'll rebase, which should pick up the place where I worked around and linked this bug (#9525)

Comment (cpcloud) [reply ibis/backends/clickhouse/compiler.py]: Yep, this should be in #9525.

**Pull request merged**